### PR TITLE
EdgeSnap grant proposal

### PR DESCRIPTION
**Pull request by ShankarWarang**
#### Project Abstract

> Please replace these instructions with a brief description of your project summarising key points (1-2 paragraphs).
>
> If your application is a follow-up to a previous grant, please mention which one in the first line of the abstract and include a link to previous pull requests if applicable.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [ ] **Level 2**:  Up to $30,000, 3 approvals
- [x] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (bank details via email _or_ Polkadot (USDC & USDT) address in the application).
- [x] I am aware that, in order to receive a grant, I (and the entity I represent) have to successfully complete a KYC/KYB check.
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@shankarwarang:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Comment by ShankarWarang** [comment]
I have read and hereby sign the Contributor License Agreement.

**Review by Raj-RR1 — COMMENTED** [review]
LGTM

**Review by takahser — CHANGES_REQUESTED** [review]
@ShankarWarang thanks for the submission of your proposal.
I have some initial inquiries:

- could you add the [mandatory deliverables](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md#milestone-1-example--basic-functionality) to all milestones? If for some reason you want to skip some of them, please specify why you want to skip them.
- the description of Chainsafe's [metamask-snap-polkadot](https://github.com/ChainSafe/metamask-snap-polkadot) repo infers compatibility with other Substrate-based chains. Is this a misconception?
image
- the https://github.com/edgeware-network/substrate-snap, https://github.com/edgeware-network/metamask-substrate-adapter, https://github.com/edgeware-network/metamask-substrate-types repos that you've linked are either private or doesn't exist
image

File: applications/EdgeSnap.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+- **Costs:** 28,500 USDT
+
+In milestone 2, we are looking to achieve seamless integration of approximately half of the total Substrate chains.
+
+| Number | Deliverable | Specification |
+| --- | --- | --- |
+| **0a.** | License | Apache 2.0 / GPLv3 / MIT / Unlicense |
+| --- | --- | --- |
+| **0b.** | UpdateDocumentation | Update the docs with newly added chains' info. |
+| 1. | Chains' Integration | Do integration for approx half (45-55) of the total chains. {Only the relaychains, parachains and solochains available on polkadot-js will be integrated.} |
```
Comment (PieWol) [inline applications/EdgeSnap.md:155]: Does this deliverable mean that you are going to create the snaps for all the (45-55) chains yourself? Integrating sounds a lot like you will have to rely on third parties to be able to deliver this milestone.

**Review by PieWol — CHANGES_REQUESTED** [review]
I left some comments, please adjust the application accordingly.

File: applications/EdgeSnap.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+• **Contact Email:** shankar@edgeware.community
+
+• **Alternative Contact Email:** gagan@edgeware.community
+
+• **Telegram:** @gmeister7
+
+### **Legal Structure**
+
+We do not have a legal structure as we are contributors to a DAO.
```
[the remark this answers is not in the document]
Comment (semuelle) [reply applications/EdgeSnap.md:74]: To clarify, a legal structure isn't strictly necessary, but then it'll be necessary to list all beneficiaries of the grant and each one needs to perform KYC.

**Review by semuelle — COMMENTED** [review]
Thanks for the application. Two comments:

- [We don't offer upfront payments](https://grants.web3.foundation/docs/faq#can-i-get-an-upfront-payment).
- Are you aware of ChainSafe's intentions for including parachains in their snap?

**Comment by ShankarWarang** [comment]
1. Thanks for the feedback @takahser. We have updated the proposal as per the suggestion.
2. Currently the [Polkadot Snap by ChainSafe](https://snaps.metamask.io/snap/npm/chainsafe/polkadot-snap/) only supports Polkadot, Kusama and Westend. Also, EdgeSnap is not just about developing snap apps for substrate chains. We are building an App Store for Substrate Snaps. Also, the major proposition here is the whole service rather than just the product. We want to accelerate the snap-readiness in the Polkadot ecosystem dApps.
3. Those repos that we've linked currently do not exist as of now and we will be storing the custom npm packages in them. At the moment, we are using the EdgeSnap repo as the mono repo.

File: applications/EdgeSnap.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ShankarWarang) [reply applications/EdgeSnap.md:74]: We are okay with doing individual KYCs.

File: applications/EdgeSnap.md (hunk 1 above)
Comment (ShankarWarang) [reply applications/EdgeSnap.md:157]: Yes! We do not rely on any third parties and all the (tech/ non-tech) deliverables will be delivered by us.

**Comment by ShankarWarang** [comment]
> Thanks for the application. Two comments:
> 
> * [We don't offer upfront payments](https://grants.web3.foundation/docs/faq#can-i-get-an-upfront-payment).
> * Are you aware of ChainSafe's intentions for including parachains in their snap?

- Noted and made the changes accordingly.
- Yes. However, we will be also including the solochains along with the parachains and relaychains (which are aligned with the Polkadot's vision, i.e. only those solochains which are available on Polkadot-JS/Apps). EdgeSnap is more of a service than just a project for the development of substrate snap apps and the brand new Substrate Snap Store. We started building on top of what ChainSafe built. ChainSafe is the core developing entity for bringing the Snap to Substrate in the first place. While we are aiming to become the accelerator for snap-readiness.

**Comment by PieWol** [comment]
Hey @ShankarWarang , I want to follow up on this reply

> 2. Currently the [Polkadot Snap by ChainSafe]
(https://snaps.metamask.io/snap/npm/chainsafe/polkadot-snap/) only supports Polkadot, Kusama and Westend. Also, EdgeSnap is not just about developing snap apps for substrate chains. We are building an App Store for Substrate Snaps. Also, the major proposition here is the whole service rather than just the product. We want to accelerate the snap-readiness in the Polkadot ecosystem dApps.

The official [metamask snap store](https://snaps.metamask.io/) is already such a store you envision with a lot of snaps available and a polished ux. What do you think will be the benefit of building another snap store? I suggest getting directly in touch with Chainsafe to see if you can simply work together on this project as it would be a pity to have two teams built on the same open source software with the same framework at the same time and most likely produce duplicates. 

On another note: Having a milestone overly expensive to then argue that a non-paid service time will follow is nothing else but upfront payment. Please adjust your milestones so that you first deliver the service or software and upon delivery you will be paid. We can't hold anybody liable in case they completed all milestones and got paid but then don't (for any reason) deliver on further free promises that might have influenced the foundation to accept a grant in the first place. 

If you really want to do something like this I would suggest to first write an application for the milestones you have, reduce the cost, and in case this project is a great success with lots of interest by ecosystem teams you can still apply for a maintenance grant. I think this makes it easier for all of us as there is no way to guess the maintenance and advisory workload that might occur in that timeframe. 

Sadly the foundation doesn't fund market outreach, feedback collection, marketing or any community events that you are planning in the scope of this grant. Please see here: [What activities/positions do you fund?](https://grants.web3.foundation/docs/faq#what-activitiespositions-do-you-fund)

I'm looking forward to hearing back from you and hope that you will be able to reach out to Chainsafe.

**Comment by ShankarWarang** [comment]
@PieWol Thanks for the follow-up.

The official [metamask snap store](https://snaps.metamask.io/) is not really what we envisioned. It only has two snaps from the whole substrate ecosystem! There could be entry barriers financially and reputation-wise. Building something substrate-native will massively benefit small/new parachains and solochains. The vercel preview (MVP) is for showcasing the functionalities. UI/UX improvements are part of the milestone 1.

ChainSafe proposed [SubstrateSnap](https://github.com/w3f/Grants-Program/blob/master/applications/maintenance/Substratesnap_Maintenance.md) almost a year ago and it's not yet available. Also, parachains/solochains support is not included in their grant. 

Snap Store is just one component of the EdgeSnap along with the Interactive UI, combined with the service which includes the development of snap apps for all the substrate chains and periodic snap integrations on Polkadot/substrate ecosystem dApps.

We are happy to remove the mention of non-paid service (maintenance for 9 months) from the proposal to avoid the influence bias. The current milestones' costs are not influenced by the free 9-month maintenance period. 

The per-chain integration costs here are substantially less than the industry average. The average total cost involved per chain for EdgeSnap considering all the milestones will be 1165.0485 USD and the average integration cost per chain (considering milestones 2 & 3) will be 728.1553 USD.

**Comment by semuelle** [comment]
> There could be entry barriers financially and reputation-wise.

IIRC, an audit is required for any snap to be listed in the Snap store. 

> ChainSafe proposed [SubstrateSnap](https://github.com/w3f/Grants-Program/blob/master/applications/maintenance/Substratesnap_Maintenance.md?rgh-link-date=2024-02-12T16%3A22%3A17Z) almost a year ago and it's not yet available. Also, parachains/solochains support is not included in their grant.

This grant was completed and the result was the initial version of https://github.com/ChainSafe/metamask-snap-polkadot. 

Given the size of the undertaking and this grant and the relative novelty of the project, I would recommend that you apply with the [Decentralized Futures Program](https://futures.web3.foundation/) instead.

**Comment by ShankarWarang** [comment]
@semuelle 

> IIRC, an audit is required for any snap to be listed in the Snap store.

Thanks for highlighting that.

> This grant was completed and the result was the initial version of https://github.com/ChainSafe/metamask-snap-polkadot.

I meant to imply that PolkadotApps/dotapps do not support the snap yet and the inclusion of parachains is not stated anywhere.

> Given the size of the undertaking and this grant and the relative novelty of the project, I would recommend that you apply with the [Decentralized Futures Program](https://futures.web3.foundation/) instead.

Does the Decentralized Futures Program process the applications on a rolling basis?

**Comment by PieWol** [comment]
@ShankarWarang it's stated on the website of the decentralized futures program, that applications are open until june 2024. So you would have a good amount of time left to apply. It's also stated that applications are reviewed on a rolling basis, yes.

**Comment by keeganquigley** [comment]
Thanks for your thorough answers @ShankarWarang and for the deep dive @PieWol I have to also concur that this would be a better fit for the DF program, since many of the deliverables cover parts of the development cycle we can't fund, and since the MVP has already been developed. For example, we can't fund feedback & analysis, weekly dev cohorts, etc. Let us know if this sounds good to you or if you'd rather refactor this application with a reduced scope/cost.

**Comment by keeganquigley** [comment]
Sounds good @ShankarWarang thanks for confirming with us. I will go ahead and close this, let us know if you have further questions.

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/edgesnap-substrate-snap-readiness-accelerator/6326/1

**Comment by ShankarWarang** [comment]
We applied for the decentralised futures grant and also received a response from David Hawig. We replied to him using the edgeware.community custom domain email. As David might not have received any email from that domain before and as it also had embedded links, we suspect that email might have ended up in the spam box. 
Can anyone please relay this to David if possible? @takahser @PieWol @semuelle @keeganquigley 
Thanks!

**Pull request closed without merging**